tests/cupyx_tests/scipy_tests/ndimage_tests/test_filters.py

```diff
@@ -12,150 +11,335 @@
+            if not footprint.any():
+                footprint = xp.ones(kshape)
+            return None, footprint
+
+        if self.filter in ('minimum_filter1d', 'maximum_filter1d'):
+            return self.ksize
+
+        raise RuntimeError('unsupported filter name')
+
+    @property
+    def _dtype(self):
+        return getattr(self, 'wdtype', None) or self.dtype
+
+    @property
+    def _ndim(self):
+        return getattr(self, 'ndim', len(getattr(self, 'shape', [])))
+
+    @property
+    def _kshape(self):
+        return getattr(self, 'kshape', (self.ksize,) * self._ndim)
+
+    @property
+    def _origin(self):
+        origin = getattr(self, 'origin', 0)
+        if origin is not None:
+            return origin
+        is_1d = self.filter.endswith('1d')
+        return -1 if is_1d else (-1, 1, -1, 1)[:self._ndim]
+
+
+# Parameters common across all modes (with some overrides)
+COMMON_PARAMS = {
+    'shape': [(4, 5), (3, 4, 5), (1, 3, 4, 5)],
+    'ksize': [3, 4],
+    'dtype': [numpy.int32, numpy.float64],
+}
+
+
+# The bulk of the tests are done with this class
 @testing.parameterize(*(
-    testing.product({
-        'shape': [(3, 4), (2, 3, 4), (1, 2, 3, 4)],
-        'ksize': [3, 4],
-        'mode': ['reflect'],
-        'cval': [0.0],
-        'origin': [0, 1, None],
-        'adtype': [numpy.int8, numpy.int16, numpy.int32,
-                   numpy.float32, numpy.float64],
-        'wdtype': [None, numpy.int32, numpy.float64],
-        'output': [None, numpy.int32, numpy.float64],
-        'filter': ['convolve', 'correlate']
-    }) + testing.product({
-        'shape': [(3, 4), (2, 3, 4), (1, 2, 3, 4)],
-        'ksize': [3, 4],
-        'mode': ['constant'],
-        'cval': [-1.0, 0.0, 1.0],
-        'origin': [0],
-        'adtype': [numpy.int32, numpy.float64],
-        'wdtype': [None],
-        'output': [None],
-        'filter': ['convolve', 'correlate']
-    }) + testing.product({
-        'shape': [(3, 4), (2, 3, 4), (1, 2, 3, 4)],
-        'ksize': [3, 4],
-        'mode': ['nearest', 'mirror', 'wrap'],
-        'cval': [0.0],
-        'origin': [0],
-        'adtype': [numpy.int32, numpy.float64],
-        'wdtype': [None],
-        'output': [None],
-        'filter': ['convolve', 'correlate']
-    })
+    testing.product([
+        # Filter-function specific params
+        testing.product({
+            'filter': ['convolve', 'correlate'],
+        }) + testing.product({
+            'filter': ['convolve1d', 'correlate1d',
+                       'minimum_filter1d', 'maximum_filter1d'],
+            'axis': [0, 1, -1],
+        }) + testing.product({
+            'filter': ['minimum_filter', 'maximum_filter'],
+            'footprint': [False, True],
+        }),
+
+        # Mode-specific params
+        testing.product({
+            **COMMON_PARAMS,
+            'mode': ['reflect'],
+            # With reflect test some of the other parameters as well
+            'origin': [0, 1, None],
```

Comment:
I suggest restricting shape for reflect only in order to cut down on the number of combinations that need to be tested. Just testing the other shapes for the rest of the modes should be good enough.

(This change reduced the number of tests from 7696 to 4500 when I tried it).

```suggestion
            'origin': [0, 1, None],
            'shape': [(4, 5)],
```

Reply:
'reflect' was being used to test all the combinations, the origin ones actually really put the multiple dimensions to the test. I would be more comfortable dropping some of the shapes from the other modes.

Reply:
I would be happier with reducing the `dtype` list 1 line later since that is tested similarly later. It is also longer so will reduce the tests even more.

Reply:
When removing the `dtype` there and also adding a F-ordered tests later it is down to 4944 (with 1000 skipped due to redundancies).